Dockerfile

```diff
@@ -15,12 +15,18 @@ RUN addgroup --gid 2000 --system appgroup && \
 WORKDIR /app
 
 RUN apt-get update && \
-    apt-get upgrade -y
+    apt-get upgrade -y && \
+    apt-get autoremove -y && \
+    rm -rf /var/lib/apt/lists/*
 
 # Stage: build assets
 FROM base as build
 
-RUN apt-get install -y make python g++
+ARG BUILD_NUMBER=1_0_0
+ARG GIT_REF=not-available
+
+RUN apt-get update && \
```

Comment:
how come you've got an `update` here but not  an `upgrade`?

Reply:
Because this is stage that is thrown away - essentially all we need is to pull in the package lists so we can install the build dependancies (make python g++).

Reply:
isn't the `update` superfluous in that we did an `update` in `base` anyway

Reply:
why do we need to do an extra one here?

Reply:
> isn't the `update` superfluous in that we did an `update` in `base` anyway

No, because we cleaned up the package lists with `rm -rf /var/lib/apt/lists/*` at line 20 - this is to save space in the final image (approx 20MB).  Previously this clean up command was in the final stage - however that was pointless in terms of saving space in the image.

The `update` here will again pull down the package lists (maybe wasting a bit of time) but we dont need to clean up because this stage is thrown away.